buildSrc/src/main/kotlin/ReleaseScripts.kt

```diff
@@ -53,7 +53,7 @@ private fun Semver.getGitTag(): String = if (build.isNotEmpty()) build.joinToStr
 data class KotlinVersion(val javaVersion: Semver, val kotlinMinor: Int, val isSnapshot: Boolean) {
     companion object {
         fun fromVersionString(versionString: String): KotlinVersion {
-            val versionStringSegments = "^(\\d+.\\d+.\\d+).(\\d+)(\\-.*\\+[\\w\\d]+)?(\\-SNAPSHOT)?\$"
+            val versionStringSegments = "^(\\d+.\\d+.\\d+).(\\d+)(-.*\\+\\w+)?(-SNAPSHOT)?\$"
```

Comment:
1. I didn't notice it before, but dots (`.`) here are not escaped, so this regex might match this string: "4x10a0z1-alpha.1674847484+ffd6999e"

1. It's always hard to read regexes, but IMO we can improve this one by using raw Kotlin strings: 

`"""^(\d+\.\d+\.\d+)\.(\d+)(-.*\+\w+)?(-SNAPSHOT)?$"""`

Reply:
Good catch!